services/ui-src/src/components/CertificationTab/CertificationTab.test.jsx

```diff
@@ -20,88 +18,103 @@ const mockUser = {
     }
   ]
 };
+
 jest.mock("../../utility-functions/userFunctions", () => ({
   getUserInfo: () => Promise.resolve(mockUser)
 }));
+
 jest.mock("../../libs/api", () => ({
   obtainUserByEmail: () => mockUser
 }));
+
+const renderWithStatus = (statusData) => {
+  const form = JSON.parse(JSON.stringify(CertificationTabMock));
+  form.currentForm.statusData = {
+    ...form.currentForm.statusData,
+    ...statusData,
+  };
+  const store = mockStore(form);
+  return render(
+    <Provider store={store}>
+      <CertificationTab />
+    </Provider>
+  );
+};
+
+const inProgress = {
+  status_id: 2,
+  status: "In Progress",
+};
+
+const provisional = {
+  status_id: 3,
+  status: "Provisional Data Certified and Submitted",
+};
+
+const final = {
+  status_id: 4,
+  status: "Final Data Certified and Submitted",
+};
+
 describe("Test CertificationTab.js", () => {
-  let wrapper;
-  let wrapper2;
-  let wrapper3;
-  let store = mockStore(CertificationTabMock); // form status = in_progress
-  let store2 = mockStore(CertificationTabMock2); // form status = provisional
-  let store3 = mockStore(CertificationTabMock3); // form status = final
-
-  beforeEach(() => {
-    wrapper = mount(
-      <Provider store={store}>
-        <CertificationTab />
-      </Provider>
-    );
-
-    wrapper2 = mount(
-      <Provider store={store2}>
-        <CertificationTab />
-      </Provider>
-    );
-
-    wrapper3 = mount(
-      <Provider store={store3}>
-        <CertificationTab />
-      </Provider>
-    );
+  it("should not display status text for in-progress forms", () => {
+    renderWithStatus(inProgress);
+    expect(screen.queryByTestId("statusText")).not.toBeInTheDocument();
   });
 
-  test("Check for modified_on and modified_by for all three statuses", async () => {
-    expect(wrapper.find({ "data-testid": "statusText" })).toEqual({});
+  it("should display status text for provisional forms", () => {
```

Comment:
these tests are very close to the setup for a describe.each setup, idk if it is worth converting, but worth pointing out if you haven't used it before

Reply:
Today I learned about `describe.each`, thank you! And I don't like it. Maaaybe someday I could see myself using it - if I had a set of tests more like [the validation tests in MCR](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/main/services/ui-src/src/utils/validation/schemas.test.ts).